src/Token.php

```diff
@@ -48,6 +48,11 @@ public function isMinimumTimeBefore(DateTimeInterface $now): bool;
      */
     public function isExpired(DateTimeInterface $now): bool;
 
+    /**
+     * Returns if the token has the claim and the value matches
+     */
+    public function withClaim(string $claimId, string $claimValue): bool;
```

Comment:
```suggestion
    public function hasClaimWithValue(string $claimId, string $claimValue): bool;
```
Not exactly sure what's the best name for this method, but `with` has been used by PSR to convey a factory method: https://www.php-fig.org/psr/psr-7/

Reply:
Of course. I'll change both token and constraint.